packages/node/src/edge-functions/edge-handler-template.js

```diff
@@ -1,49 +1,22 @@
 // provided by the edge runtime:
 /* global addEventListener */
 
-function buildUrl(requestDetails) {
-  const host = requestDetails.headers['x-forwarded-host'] || '127.0.0.1';
-  const path = requestDetails.url || '/';
-
-  const allProtocols = requestDetails.headers['x-forwarded-proto'];
-  let proto;
-  if (allProtocols) {
-    // handle multi-protocol like: https,http://...
-    proto = allProtocols.split(/\b/).shift();
-  } else {
-    proto = 'http';
-  }
-
-  return `${proto}://${host}${path}`;
+function getUrl(url, host) {
+  const urlObj = new URL(url);
+  urlObj.host = host;
+  return urlObj.toString();
 }
 
-async function respond(
-  userEdgeHandler,
-  requestDetails,
-  event,
-  options,
-  dependencies
-) {
+async function respond(userEdgeHandler, event, options, dependencies) {
   const { Request, Response } = dependencies;
   const { isMiddleware } = options;
 
-  let body;
-
-  if (requestDetails.method !== 'GET' && requestDetails.method !== 'HEAD') {
-    if (requestDetails.body) {
-      body = Uint8Array.from(atob(requestDetails.body), c => c.charCodeAt(0));
-    }
-  }
-
-  const request = new Request(buildUrl(requestDetails), {
-    headers: requestDetails.headers,
-    method: requestDetails.method,
-    body: body,
-  });
-
-  event.request = request;
-
-  let response = await userEdgeHandler(event.request, event);
+  const host = event.request.headers.get('x-forwarded-host');
+  event.request.headers.set('host', host);
+  let response = await userEdgeHandler(
+    new Request(getUrl(event.request.url, host), event.request),
```

Comment:
I removed that code because I didn't see any test using it more than the unit test around the template. 

What could be an e2e that demonstrates that logic is still necessary?

Reply:
I was having trouble writing an e2e test for it, but it does solve the problem the user had. I think we should restore the functionality and unit tests, at least.

Reply:
I still think this is a regression